### PR TITLE
[Feat] Upgrade deck.gl@8.4.11 luma.gl@8.4.3 loaders.gl@2.3.12

### PR DESCRIPTION
**Pull request by heshan0131**
* fix grid and hexagon layer, added tests
* added `test-headless` and moved map-container tests there
Signed-off-by: Shan He <heshan0131@gmail.com>

[Bug] Fix cpu-aggregator can't access index of undefined 

Signed-off-by: Shan He <heshan0131@gmail.com>

[Bug] Pinning a tooltip moves it to the top left of the screen

* fix grid and hexagon layer, added tests
* added `test-headless` and moved map-container tests there

### CODE REVIEW

File: src/layers/aggregation-layer.js [hunk 1]
```diff
@@ -248,18 +255,25 @@ export default class AggregationLayer extends Layer {
...
-    const getElevationValue = getValueAggrFunc(
+    const getElevationValue = aggregatePoints(
       this.config.sizeField,
       this.config.visConfig.sizeAggregation
     );
     const hasFilter = Object.values(gpuFilter.filterRange).some(arr => arr.some(v => v !== 0));
 
-    const getFilterValue = gpuFilter.filterValueAccessor(dataContainer)();
+    // TODO: fix fr data container
+    // const getFilterValue = gpuFilter.filterValueAccessor(dataContainer)();
```
Comment (igorDykhta) [inline src/layers/aggregation-layer.js:271]: Could you give me an example when this doesn't work?

File: src/layers/index.d.ts [hunk 2]
```diff
@@ -93,7 +93,7 @@ export class Layer {
   getVisualChannelDescription(key: string): VisualChannelDescription;
   isLayerHovered(objectInfo: any): boolean;
   hasHoveredObject(objectInfo: any): any | null;
-  getHoverData(object: any, allData?: KeplerTable['allData'], fields?: KeplerTable['fields']): any;
+  getHoverData(object: any, dataContainer?: KeplerTable['allData'], fields?: KeplerTable['fields']): any;
```
Comment (igorDykhta) [inline src/layers/index.d.ts:96]: KeplerTable['allData'] ?

**Review by igorDykhta — APPROVED** [review]
@heshan0131 I added fixes for line altitude and aggregation layers.
Did some testing after that, looks fine.

I've only managed to somehow make hovering not working one time, but cant repro again.

File: src/components/common/radius-legend.js [hunk 3]
```diff
@@ -0,0 +1,169 @@
+// Copyright (c) 2022 Uber Technologies, Inc.
```
Comment (heshan0131) [inline src/components/common/radius-legend.js:1]: Can you remove this file from this PR?

**Pull request merged**